doc/source/reference/arrays.dtypes.rst

```diff
@@ -178,12 +178,18 @@ Built-in Python types
     :class:`bool`     :class:`bool\_`
     :class:`float`    :class:`float\_`
     :class:`complex`  :class:`cfloat`
-    :class:`str`      :class:`string`
+    :class:`bytes`    :class:`bytes\_`
+    :class:`str`      :class:`bytes\_` (Python2) or :class:`unicode\_` (Python3)
     :class:`unicode`  :class:`unicode\_`
     :class:`buffer`   :class:`void`
     (all others)      :class:`object_`
     ================  ===============
 
+    Note that ``str`` refers to either null terminated bytes or unicode strings
```

Comment:
If we are talking about array types, the unicode strings are null terminated  UTF-32.